packages/turf-boolean-within/index.js

```diff
@@ -0,0 +1,218 @@
+var inside = require('@turf/inside');
+var calcBbox = require('@turf/bbox');
+var invariant = require('@turf/invariant');
+var isPointOnLine = require('@turf/boolean-point-on-line');
+var getGeom = invariant.getGeom;
+var getGeomType = invariant.getGeomType;
+
+/**
+ * Boolean-within returns true if the first geometry is completely within the second geometry.
+ * The interiors of both geometries must intersect and, the interior and boundary of the primary (geometry a)
+ * must not intersect the exterior of the secondary (geometry b).
+ * Boolean-within returns the exact opposite result of the `@turf/boolean-contains`.
+ *
+ * @name booleanWith
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {boolean} true/false
+ * @example
+ * const line = turf.lineString([[1, 1], [1, 2], [1, 3], [1, 4]]);
+ * const point = turf.point([1, 2]);
+ *
+ * turf.booleanWithin(point, line);
+ * //=true
+ */
+module.exports = function (feature1, feature2) {
+    var type1 = getGeomType(feature1);
+    var type2 = getGeomType(feature2);
+    var geom1 = getGeom(feature1);
+    var geom2 = getGeom(feature2);
+
+    switch (type1) {
+    case 'Point':
+        switch (type2) {
+        case 'MultiPoint':
+            return isPointInMultiPoint(geom1, geom2);
+        case 'LineString':
+            return isPointOnLine(geom1, geom2, true);
+        case 'Polygon':
+            return inside(geom1, geom2, true);
+        default:
+            throw new Error('feature2 ' + type2 + ' geometry not supported');
+        }
+    case 'MultiPoint':
+        switch (type2) {
+        case 'MultiPoint':
+            return isMultiPointInMultiPoint(geom1, geom2);
+        case 'LineString':
+            return isMultiPointOnLine(geom1, geom2);
+        case 'Polygon':
+            return isMultiPointInPoly(geom1, geom2);
+        default:
+            throw new Error('feature2 ' + type2 + ' geometry not supported');
+        }
+    case 'LineString':
+        switch (type2) {
+        case 'LineString':
+            return isLineOnLine(geom1, geom2);
+        case 'Polygon':
+            return isLineInPoly(geom1, geom2);
+        default:
+            throw new Error('feature2 ' + type2 + ' geometry not supported');
+        }
+    case 'Polygon':
+        switch (type2) {
+        case 'Polygon':
+            return isPolyInPoly(geom1, geom2);
+        default:
+            throw new Error('feature2 ' + type2 + ' geometry not supported');
+        }
+    default:
+        throw new Error('feature1 ' + type1 + ' geometry not supported');
```

Comment:
No support for MultiPolygon or MultiLineString?

Reply:
Not yet, but there probably should be.  The problem is there aren't any picture for it and so I always forget :)

Reply:
HAHA no pictures 🤣 Those pictures are so helpful!